t/03-auth.t

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use Mojo::Base -strict;
+
+use FindBin;
+use lib "$FindBin::Bin/lib";
+use Test::MockModule;
+use Test::More;
+use Test::Mojo;
+use Test::Warnings;
+use OpenQA::Test::Database;
+use Mojo::File qw(tempdir path);
+
+my $t;
+my $tempdir = tempdir;
+
+sub test_auth_method_startup {
+    my $auth = shift;
+    my @conf = ("[auth]\n", "method = \t  $auth \t\n");
+    $ENV{OPENQA_CONFIG} = $tempdir;
+    $tempdir->child("openqa.ini")->spurt(@conf);
+
+    $t = Test::Mojo->new('OpenQA::WebAPI');
+    ok($t->app->config->{auth}->{method} eq $auth, "started successfully with auth $auth");
```

Comment:
```suggestion
    is($t->app->config->{auth}->{method}, $auth, "started successfully with auth $auth");
```

Reply:
Did you check if the subroutine needs `$Test::Builder::Level` adjustment to report a failure correctly?

Reply:
I have implemented your change from "ok" to "is" even though I will put that into another commit as I just moved out the code.

For the "Level" change I would not do that here because then you have errors like

```
not ok 1 - started successfully with auth Fake
#   Failed test 'started successfully with auth Fake'
#   at t/03-auth.t line 44.
#          got: 'Fake'
#     expected: 'Fakefoo'
```

so you know where the helper method was triggered but you don't know what actual check failed. In cases like these I recommend to keep the "level" and where necessary either follow the flow of the test with output of previous tests that point to the calling location or output a full stack.